src/mjolnir/osmaccessrestriction.cc

```diff
@@ -7,13 +7,6 @@ using namespace valhalla::baldr;
 namespace valhalla {
 namespace mjolnir {
 
-OSMAccessRestriction::OSMAccessRestriction() {
-  memset(this, 0, sizeof(OSMAccessRestriction));
```

Comment:
Fixes 
```
valhalla/src/mjolnir/osmaccessrestriction.cc:11:10: error: first argument in call to 'memset' is a pointer to non-trivially copyable type 'valhalla::mjolnir::OSMAccessRestriction' [-Werror,-Wnontrivial-memcall]
   11 |   memset(this, 0, sizeof(OSMAccessRestriction));
      |          ^
/Users/kinkard/Developer/valhalla/src/mjolnir/osmaccessrestriction.cc:11:10: note: explicitly cast the pointer to silence this warning
   11 |   memset(this, 0, sizeof(OSMAccessRestriction));
      |          ^
      |          (void*)
1 error generated.
```

Reply:
frankly i always hated this anyway. we have some many structs that should be header only with no setters/getters. apart from the casting they literally do nothing. its just bloat for no purpose